microcosm_pubsub/backoff.py

```diff
@@ -0,0 +1,66 @@
+"""
+Consumer backoff policy
+
+"""
+from abc import ABCMeta, abstractmethod
+from random import randint
+from six import add_metaclass
+
+
+# we cannot go over 12 hours
+MAX_BACKOFF_TIMEOUT = 60 * 60 * 12
+
+
+@add_metaclass(ABCMeta)
+class BackoffPolicy(object):
+
+    def __init__(self, visibility_timeout_seconds=None):
+        self.visibility_timeout_seconds = visibility_timeout_seconds
+
+    @abstractmethod
+    def compute_backoff_timeout(self, message, message_timeout):
+        pass
+
+    @classmethod
+    def choose_backoff_policy(cls, name):
+        for subclass in cls.__subclasses__():
+            if subclass.__name__ == name:
+                return subclass
+        raise Exception("No backoff policy configured with class name: {}".format(name))
+
+
+class NaiveBackoffPolicy(BackoffPolicy):
```

Comment:
This implements our current behavior. It either takes the default backoff or the message specific one. If neither are set, no backoff is applied and the queue's default will be used (30s?).